### ISSUE TITLE
Create objects once instead of using singletons

### ISSUE THREAD

**Issue by tomjn** [royboy789/gutenberg-object-plugin#10]
This:

```php
class GutenbergObjectPlugin {
	use Singleton;
	public function __construct() {
		Scripts::init();
		Database::init();
		API::init();
		Hooks::init();
	}
}
$GutesArraySave = GutenbergObjectPlugin::init();
```

Is nigh impossible to write unit tests for.

Instead, eliminate the singletons by doing something like this:

```php
$scripts = new Scripts();
$database = new Database();
$api = new API();
$hooks = new Hooks(); 
$GutesArraySave = new GutenbergObjectPlugin( $scripts, $database, $api, $hooks );
$GutesArraySave->init();
```

This way the classes are fully testable, and rely on dependency injection to gain access to things, rather than reaching out magically across the program like a `goto` statement, with hidden global state and mytery sauce semantics.

Additionally, this provides structure to the program, and an object graph. E.g. we can now swap out the `Database` object for a different object with the same interface, or reuse that object. Possibilities!

**Comment by royboy789** [comment]
hmm. I like it! 

will work on that.

**Comment by royboy789** [comment]
Closed with #11
